oshi-core/src/main/java/oshi/util/ExecutingCommand.java

```diff
@@ -116,16 +116,39 @@ private ExecutingCommand() {
      *         string if the command failed
      */
     public static List<String> runNative(String[] cmdToRunWithArgs, String[] envp) {
+        Process p = null;
         try {
-            Process p = Runtime.getRuntime().exec(cmdToRunWithArgs, envp);
-            return getProcessOutputAndDestroy(p, cmdToRunWithArgs);
+            p = Runtime.getRuntime().exec(cmdToRunWithArgs, envp);
+            return getProcessOutput(p, cmdToRunWithArgs);
         } catch (SecurityException | IOException e) {
             LOG.trace("Couldn't run command {}: {}", Arrays.toString(cmdToRunWithArgs), e.getMessage());
+        } finally {
+            if (p != null) {
+                p.destroy();
+                if (p.getOutputStream() != null) {
```

Comment:
I noticed in the JDK source they never null checked `getOutputStream()`.  If `p` was not null they jumped straight to `close()`.  Reading the [`javadocs`](https://docs.oracle.com/javase/8/docs/api/java/lang/Process.html#getInputStream--) it states that either the stream will be returned, or if there is none:
> ... this method will return a null input stream.

A "null input stream" is not null; it's a valid `InputStream` object for which `write()` gives an exception and `close()` is a no-op.   The same is true for the output and error streams.    So it is unnecessary to check for a null input stream here.  The JDK test case (for Solaris) above just does this:
```
p.destroy();
p.getInputStream().close();
p.getErrorStream().close();
```
or, in another place
```
p.getInputStream().close();
p.getErrorStream().close();
p.destroy();
```
So I'd suggest you do the same thing here.  Basically, for the finally block:
1. null check p.  if p is not null
2. if solaris, close streams (no null check required but do check for exception on close)
3. for all platforms, destroy